[BUG] Build fails when following install from source instructions.

When trying to follow the instructions at https://github.com/deepmodeling/deepmd-kit/blob/master/doc/install/install-from-source.md the build fails both on the macOS and linux machines I have tested it on. 

After following the instructions up to `pip install .` I get the following stack trace:

```bash
      -- Build files have been written to: /Users/comprhys/deepmd-kit/_skbuild/macosx-11.0-x86_64-3.8/cmake-build
      [1/55] Building CXX object op/CMakeFiles/op_abi.dir/ewald_recp.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/ewald_recp.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/ewald_recp.cc.o -MF op/CMakeFiles/op_abi.dir/ewald_recp.cc.o.d -o op/CMakeFiles/op_abi.dir/ewald_recp.cc.o -c /Users/comprhys/deepmd-kit/source/op/ewald_recp.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/ewald_recp.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      In file included from /Users/comprhys/deepmd-kit/source/op/ewald_recp.cc:2:
      /Users/comprhys/deepmd-kit/source/lib/include/ewald.h:5:9: fatal error: 'omp.h' file not found
      #include<omp.h>
              ^~~~~~~
      3 errors generated.
      [2/55] Building CXX object op/CMakeFiles/op_abi.dir/custom_op.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/custom_op.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/custom_op.cc.o -MF op/CMakeFiles/op_abi.dir/custom_op.cc.o.d -o op/CMakeFiles/op_abi.dir/custom_op.cc.o -c /Users/comprhys/deepmd-kit/source/op/custom_op.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [3/55] Building CXX object op/CMakeFiles/op_abi.dir/map_aparam.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/map_aparam.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/map_aparam.cc.o -MF op/CMakeFiles/op_abi.dir/map_aparam.cc.o.d -o op/CMakeFiles/op_abi.dir/map_aparam.cc.o -c /Users/comprhys/deepmd-kit/source/op/map_aparam.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/map_aparam.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [4/55] Building CXX object op/CMakeFiles/op_abi.dir/prod_virial.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/prod_virial.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/prod_virial.cc.o -MF op/CMakeFiles/op_abi.dir/prod_virial.cc.o.d -o op/CMakeFiles/op_abi.dir/prod_virial.cc.o -c /Users/comprhys/deepmd-kit/source/op/prod_virial.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/prod_virial.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [5/55] Building CXX object op/CMakeFiles/op_abi.dir/prod_force_multi_device.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/prod_force_multi_device.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/prod_force_multi_device.cc.o -MF op/CMakeFiles/op_abi.dir/prod_force_multi_device.cc.o.d -o op/CMakeFiles/op_abi.dir/prod_force_multi_device.cc.o -c /Users/comprhys/deepmd-kit/source/op/prod_force_multi_device.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/prod_force_multi_device.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [6/55] Building CXX object op/CMakeFiles/op_abi.dir/prod_force.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/prod_force.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/prod_force.cc.o -MF op/CMakeFiles/op_abi.dir/prod_force.cc.o.d -o op/CMakeFiles/op_abi.dir/prod_force.cc.o -c /Users/comprhys/deepmd-kit/source/op/prod_force.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/prod_force.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [7/55] Building CXX object op/CMakeFiles/op_abi.dir/soft_min_force.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/soft_min_force.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/soft_min_force.cc.o -MF op/CMakeFiles/op_abi.dir/soft_min_force.cc.o.d -o op/CMakeFiles/op_abi.dir/soft_min_force.cc.o -c /Users/comprhys/deepmd-kit/source/op/soft_min_force.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/soft_min_force.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [8/55] Building CXX object op/CMakeFiles/op_abi.dir/soft_min_virial.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/soft_min_virial.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/soft_min_virial.cc.o -MF op/CMakeFiles/op_abi.dir/soft_min_virial.cc.o.d -o op/CMakeFiles/op_abi.dir/soft_min_virial.cc.o -c /Users/comprhys/deepmd-kit/source/op/soft_min_virial.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/soft_min_virial.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [9/55] Building CXX object op/CMakeFiles/op_abi.dir/pair_tab.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/pair_tab.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/pair_tab.cc.o -MF op/CMakeFiles/op_abi.dir/pair_tab.cc.o.d -o op/CMakeFiles/op_abi.dir/pair_tab.cc.o -c /Users/comprhys/deepmd-kit/source/op/pair_tab.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/pair_tab.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [10/55] Building CXX object op/CMakeFiles/op_abi.dir/prod_virial_multi_device.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/prod_virial_multi_device.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/prod_virial_multi_device.cc.o -MF op/CMakeFiles/op_abi.dir/prod_virial_multi_device.cc.o.d -o op/CMakeFiles/op_abi.dir/prod_virial_multi_device.cc.o -c /Users/comprhys/deepmd-kit/source/op/prod_virial_multi_device.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/prod_virial_multi_device.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [11/55] Building CXX object op/CMakeFiles/op_abi.dir/neighbor_stat.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/neighbor_stat.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/neighbor_stat.cc.o -MF op/CMakeFiles/op_abi.dir/neighbor_stat.cc.o.d -o op/CMakeFiles/op_abi.dir/neighbor_stat.cc.o -c /Users/comprhys/deepmd-kit/source/op/neighbor_stat.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/neighbor_stat.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [12/55] Building CXX object op/CMakeFiles/op_abi.dir/soft_min.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/soft_min.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/soft_min.cc.o -MF op/CMakeFiles/op_abi.dir/soft_min.cc.o.d -o op/CMakeFiles/op_abi.dir/soft_min.cc.o -c /Users/comprhys/deepmd-kit/source/op/soft_min.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/soft_min.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      In file included from /Users/comprhys/deepmd-kit/source/op/soft_min.cc:2:
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:787:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized std::vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:896:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:1004:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      3 warnings and 2 errors generated.
      [13/55] Building CXX object op/CMakeFiles/op_abi.dir/descrpt_se_a_ef.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/descrpt_se_a_ef.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/descrpt_se_a_ef.cc.o -MF op/CMakeFiles/op_abi.dir/descrpt_se_a_ef.cc.o.d -o op/CMakeFiles/op_abi.dir/descrpt_se_a_ef.cc.o -c /Users/comprhys/deepmd-kit/source/op/descrpt_se_a_ef.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/descrpt_se_a_ef.cc:2:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      In file included from /Users/comprhys/deepmd-kit/source/op/descrpt_se_a_ef.cc:3:
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:787:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized std::vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:896:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:1004:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      3 warnings and 2 errors generated.
      [14/55] Building CXX object op/CMakeFiles/op_abi.dir/gelu_multi_device.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/gelu_multi_device.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/gelu_multi_device.cc.o -MF op/CMakeFiles/op_abi.dir/gelu_multi_device.cc.o.d -o op/CMakeFiles/op_abi.dir/gelu_multi_device.cc.o -c /Users/comprhys/deepmd-kit/source/op/gelu_multi_device.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/gelu_multi_device.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      [15/55] Building CXX object op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_para.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_para.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_para.cc.o -MF op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_para.cc.o.d -o op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_para.cc.o -c /Users/comprhys/deepmd-kit/source/op/descrpt_se_a_ef_para.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/descrpt_se_a_ef_para.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      In file included from /Users/comprhys/deepmd-kit/source/op/descrpt_se_a_ef_para.cc:2:
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:787:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized std::vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:896:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:1004:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      3 warnings and 2 errors generated.
      [16/55] Building CXX object op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_vert.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_vert.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_vert.cc.o -MF op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_vert.cc.o.d -o op/CMakeFiles/op_abi.dir/descrpt_se_a_ef_vert.cc.o -c /Users/comprhys/deepmd-kit/source/op/descrpt_se_a_ef_vert.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/descrpt_se_a_ef_vert.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      In file included from /Users/comprhys/deepmd-kit/source/op/descrpt_se_a_ef_vert.cc:2:
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:787:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized std::vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:896:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:1004:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      3 warnings and 2 errors generated.
      [17/55] Building CXX object op/CMakeFiles/op_abi.dir/descrpt.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/descrpt.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/descrpt.cc.o -MF op/CMakeFiles/op_abi.dir/descrpt.cc.o.d -o op/CMakeFiles/op_abi.dir/descrpt.cc.o -c /Users/comprhys/deepmd-kit/source/op/descrpt.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/descrpt.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      In file included from /Users/comprhys/deepmd-kit/source/op/descrpt.cc:2:
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:787:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized std::vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:896:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /Users/comprhys/deepmd-kit/source/lib/include/ComputeDescriptor.h:1004:55: warning: expression result unused [-Wunused-value]
        assert( fabs(deepmd::dot3(ef, ef) - 1.0) < 1e-12 ), "ef should be a normalized vector";
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      3 warnings and 2 errors generated.
      [18/55] Building CXX object op/CMakeFiles/op_abi.dir/prod_env_mat_multi_device.cc.o
      FAILED: op/CMakeFiles/op_abi.dir/prod_env_mat_multi_device.cc.o
      /Library/Developer/CommandLineTools/usr/bin/c++ -Dop_abi_EXPORTS -I/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include -I/Users/comprhys/deepmd-kit/source/lib/include -std=c++11 -Wno-ignored-attributes -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.0 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT op/CMakeFiles/op_abi.dir/prod_env_mat_multi_device.cc.o -MF op/CMakeFiles/op_abi.dir/prod_env_mat_multi_device.cc.o.d -o op/CMakeFiles/op_abi.dir/prod_env_mat_multi_device.cc.o -c /Users/comprhys/deepmd-kit/source/op/prod_env_mat_multi_device.cc
      In file included from /Users/comprhys/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1:
      In file included from /Users/comprhys/deepmd-kit/source/op/custom_op.h:5:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/op.h:25:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/full_type_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/node_def_util.h:23:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/attr_value_util.h:24:
      In file included from /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor.h:25:
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:25: error: 'auto' not allowed in lambda parameter
          auto all = [](const auto&... bool_vals) {
                              ^~~~
      /Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow/include/tensorflow/core/framework/tensor_types.h:176:30: error: type 'int &' of function parameter pack does not contain any unexpanded parameter packs
          auto all = [](const auto&... bool_vals) {
                        ~~~~~~~~~~~^~~~~~~~~~~~~
      2 errors generated.
      ninja: build stopped: subcommand failed.
        File "/private/var/folders/yh/fjbfn5h91dgc6p34j72mzq_r0000gn/T/pip-build-env-5jzz9_uk/overlay/lib/python3.8/site-packages/skbuild/setuptools_wrap.py", line 595, in setup
          cmkr.make(make_args, env=env)
        File "/private/var/folders/yh/fjbfn5h91dgc6p34j72mzq_r0000gn/T/pip-build-env-5jzz9_uk/overlay/lib/python3.8/site-packages/skbuild/cmaker.py", line 609, in make
          raise SKBuildError(
      
      
      --------------------------------------------------------------------------------
      -- Trying "Ninja" generator
      --------------------------------
      ---------------------------
      ----------------------
      -----------------
      ------------
      -------
      --
      --
      -------
      ------------
      -----------------
      ----------------------
      ---------------------------
      --------------------------------
      -- Trying "Ninja" generator - success
      --------------------------------------------------------------------------------
      
      Configuring Project
        Working directory:
          /Users/comprhys/deepmd-kit/_skbuild/macosx-11.0-x86_64-3.8/cmake-build
        Command:
          cmake /Users/comprhys/deepmd-kit/source -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/Users/comprhys/deepmd-kit/_skbuild/macosx-11.0-x86_64-3.8/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/Users/comprhys/deepmd-kit/tensorflow_env/bin/python -DPYTHON_VERSION_STRING:STRING=3.8.6 -DPYTHON_INCLUDE_DIR:PATH=/Users/comprhys/.pyenv/versions/3.8.6/include/python3.8 -DPYTHON_LIBRARY:FILEPATH=/Users/comprhys/.pyenv/versions/3.8.6/lib/libpython3.8.a -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/private/var/folders/yh/fjbfn5h91dgc6p34j72mzq_r0000gn/T/pip-build-env-5jzz9_uk/overlay/lib/python3.8/site-packages/skbuild/resources/cmake -DCMAKE_MAKE_PROGRAM:FILEPATH=/private/var/folders/yh/fjbfn5h91dgc6p34j72mzq_r0000gn/T/pip-build-env-5jzz9_uk/overlay/lib/python3.8/site-packages/ninja/data/bin/ninja -DTENSORFLOW_ROOT:STRING=/Users/comprhys/deepmd-kit/tensorflow_env/lib/python3.8/site-packages/tensorflow -DBUILD_PY_IF:BOOL=TRUE -DBUILD_CPP_IF:BOOL=FALSE -DCMAKE_BUILD_TYPE:STRING=Release -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=11.0 -DCMAKE_OSX_ARCHITECTURES:STRING=x86_64
      
      Traceback (most recent call last):
      
      An error occurred while building with CMake.
        Command:
          cmake --build . --target install --config Release --
        Source directory:
          /Users/comprhys/deepmd-kit
        Working directory:
          /Users/comprhys/deepmd-kit/_skbuild/macosx-11.0-x86_64-3.8/cmake-build
      Please see CMake's output for more information.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for deepmd-kit
  Building wheel for python-hostlist (setup.py) ... done
  Created wheel for python-hostlist: filename=python_hostlist-1.21-py3-none-any.whl size=38943 sha256=73dedab60c0689e6b91f5dcf714f89dd6717b3383cee9751f7fd439e97c0e679
  Stored in directory: /Users/comprhys/Library/Caches/pip/wheels/bc/4a/f3/7fdfcaf984825c19c43122f8d1217aa6a07934f3360dc5a6a3
Successfully built python-hostlist
Failed to build deepmd-kit
ERROR: Could not build wheels for deepmd-kit, which is required to install pyproject.toml-based projects

```

--- a/source/lib/include/ewald.h
+++ b/source/lib/include/ewald.h
@@ -2,7 +2,12 @@
 
 #include<algorithm>
 #include<cassert>
+#if defined(_OPENMP)
 #include<omp.h>
+#else
+int omp_get_num_threads() { return 1; }
+int omp_get_thread_num() { return 0; }
+#endif
 
 #include "utilities.h"
 #include "region.h"

